tests/test_pyth_client.py

```diff
@@ -0,0 +1,378 @@
+    )
+
+
+@ pytest.fixture
+def watch_session(solana_client: SolanaClient) -> WatchSession:
+    return WatchSession(solana_client)
+
+
+@ pytest.fixture
+def mapping_account(solana_client: SolanaClient) -> PythMappingAccount:
+    mapping_account = PythMappingAccount(
+        key=SolanaPublicKey(V2_FIRST_MAPPING_ACCOUNT_KEY),
+        solana=solana_client
+    )
+    return mapping_account
+
+
+@ pytest.fixture
+def mapping_account_entries() -> List[SolanaPublicKey]:
+    return [
+        SolanaPublicKey(BCH_PRODUCT_ACCOUNT_KEY_DEVNET)
+    ]
+
+
+@ pytest.fixture
+def product_account(solana_client: SolanaClient) -> PythProductAccount:
+    product_account = PythProductAccount(
+        key=SolanaPublicKey(BCH_PRODUCT_ACCOUNT_KEY_DEVNET),
+        solana=solana_client,
+    )
+    product_account.slot = 96866599
+    product_account.attrs = {
+        'asset_type': 'Crypto',
+        'symbol': 'BCH/USD',
+        'quote_currency': 'USD',
+        'description': 'BCH/USD',
+        'generic_symbol': 'BCHUSD',
+        'base': 'BCH'
+    }
+    product_account.first_price_account_key = SolanaPublicKey(
+        BCH_PRICE_ACCOUNT_KEY_DEVNET,
+    )
+    return product_account
+
+
+@ pytest.fixture
+def product_account_bytes() -> bytes:
+    return base64.b64decode(PRODUCT_ACCOUNT_B64_DATA)[_ACCOUNT_HEADER_BYTES:]
+
+
+@ pytest.fixture
+def price_account(solana_client: SolanaClient) -> PythPriceAccount:
+    return PythPriceAccount(
+        key=SolanaPublicKey(BCH_PRICE_ACCOUNT_KEY_DEVNET),
+        solana=solana_client,
+    )
+
+
+@ pytest.fixture
+def price_account_bytes() -> bytes:
+    return base64.b64decode(PRICE_ACCOUNT_B64_DATA)[_ACCOUNT_HEADER_BYTES:]
+
+
+@pytest.fixture()
+def mock_get_account_info(mocker: MockerFixture) -> AsyncMock:
+    async_mock = AsyncMock(side_effect=get_account_info_resp)
+    mocker.patch('pythclient.solana.SolanaClient.get_account_info', side_effect=async_mock)
+    return async_mock
+
+
+@ pytest.fixture()
+def mock_get_program_accounts(mocker: MockerFixture) -> AsyncMock:
+    async_mock = AsyncMock(side_effect=get_program_accounts_resp)
+    mocker.patch('pythclient.solana.SolanaClient.get_program_accounts', side_effect=async_mock)
+    return async_mock
+
+
+def test_products_property_not_loaded(pyth_client: PythClient):
+    with pytest.raises(NotLoadedException):
+        pyth_client.products
+
+
+@pytest.mark.asyncio
+async def test_get_products(
+    pyth_client: PythClient,
+    mock_get_account_info: AsyncMock,
+    product_account: PythProductAccount
+) -> None:
+    products = await pyth_client.get_products()
+    for i in range(len(products)):
+        assert products[i].key == product_account.key
+
+
+def test_get_ratelimit(
+    pyth_client: PythClient,
+) -> None:
+    ratelimit = pyth_client.solana_ratelimit
+    assert ratelimit._get_overall_interval() == 0
+    assert ratelimit._get_method_interval() == 0
+    assert ratelimit._get_connection_interval() == 0
```

Comment:
Other than code coverage, what is this actually testing? You're not setting anything up really. Better would be to set the rate limits and then make assertions based on these.

Reply:
will do